core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -38,69 +42,185 @@ fun addBrakingCurvesAtEOAs(
+                context,
+                overhead,
+                targetPosition,
+                targetSpeed,
+                BrakingType.ETCS_SBD
+            )
+        assert(
+            sbdCurve.beginPos >= 0 &&
+            sbdCurve.beginSpeed == maxSpeedEnvelope &&
+                sbdCurve.endSpeed == targetSpeed &&
+                sbdCurve.endPos == targetPosition
+        )
         val guiCurve =
             computeBrakingCurve(
                 context,
                 overhead,
-                beginPos,
                 targetPosition,
                 targetSpeed,
                 BrakingType.ETCS_GUI
             )
-        val sbdCurve =
+        assert(
+            guiCurve.beginPos >= 0 &&
+            guiCurve.beginSpeed == maxSpeedEnvelope &&
+                guiCurve.endSpeed == targetSpeed &&
+                guiCurve.endPos == targetPosition
+        )
+
+        val fullIndicationCurve =
+            computeIndicationBrakingCurveFromRef(context, sbdCurve, BrakingCurveType.SBD, guiCurve)
+        assert(
+            fullIndicationCurve.beginSpeed == maxSpeedEnvelope &&
+                fullIndicationCurve.endSpeed == targetSpeed &&
+                fullIndicationCurve.endPos == targetPosition
+        )
+
+        // Indication curve for EOAs will always have at least 2 points with positive positions.
+        // This method cannot return a null.
+        val indicationCurve =
+            keepBrakingCurveUnderOverlay(fullIndicationCurve, envelope, beginPos)!!
+        assert(
+            indicationCurve.beginPos >= beginPos &&
+                indicationCurve.beginSpeed <= maxSpeedEnvelope &&
+                indicationCurve.endSpeed == targetSpeed &&
+                indicationCurve.endPos == targetPosition
+        )
+        builder.addPart(indicationCurve)
+
+        // We build EOAs along the path. We need to handle overlaps with the next EOA. To do so, we
+        // shift the left position constraint, beginPos, to this EOA's target position.
+        beginPos = targetPosition
+    }
+    return builder.build()
+}
+
+/** Compute braking curves at every limit of authority. */
+fun addBrakingCurvesAtLOAs(
+    envelope: Envelope,
+    context: EnvelopeSimContext,
+    limitsOfAuthority: Collection<LimitOfAuthority>
+): Envelope {
+    val sortedLimitsOfAuthority = limitsOfAuthority.sortedBy { it.offset }
+    val beginPos = envelope.beginPos
+    var envelopeWithLoaBrakingCurves = envelope
+    var builder = OverlayEnvelopeBuilder.forward(envelopeWithLoaBrakingCurves)
+    for (limitOfAuthority in sortedLimitsOfAuthority) {
+        val targetPosition = limitOfAuthority.offset.distance.meters
+        val targetSpeed = limitOfAuthority.speed
+        val maxSpeedEnvelope = envelopeWithLoaBrakingCurves.maxSpeed
+
+        // Add maxBecDeltaSpeed to EBD curve overlay so it reaches sufficiently high speeds to
+        // guarantee that the corresponding EBI curve intersects with envelope max speed.
```

Comment:
nit (keep only what you want, it's fine as-is if you prefer):
```suggestion
        // Add maxBecDeltaSpeed to EBD curve overlay so it reaches sufficiently high speed to
        // guarantee that, after speed translation, the corresponding EBI curve does intersect with envelope max speed.
```